frontend/src/vault/keysign/utils/getTxInputData.ts

```diff
@@ -1,75 +1,70 @@
 import { WalletCore } from '@trustwallet/wallet-core';
 
-import { getErc20ApprovePreSignedImageHashes } from '../../../chain/evm/tx/getErc20ApprovePreSignedImageHashes';
+import { getErc20ApproveTxInputData } from '../../../chain/evm/tx/getErc20ApproveTxInputData';
 import { incrementKeysignPayloadNonce } from '../../../chain/evm/tx/incrementKeysignPayloadNonce';
-import { getThorchainSwapPreSignedImageHashes } from '../../../chain/swap/native/thor/tx/getThorchainSwapPreSignedImageHashes';
-import { getOneInchSwapPreSignedImageHashes } from '../../../chain/swap/oneInch/tx/getOneInchSwapPreSignedImageHashes';
-import { getPreSigningHashes } from '../../../chain/tx/utils/getPreSigningHashes';
+import { getThorchainSwapTxInputData } from '../../../chain/swap/native/thor/tx/getThorchainSwapTxInputData';
+import { getOneInchSwapTxInputData } from '../../../chain/swap/oneInch/tx/getOneInchSwapTxInputData';
 import { KeysignPayload } from '../../../gen/vultisig/keysign/v1/keysign_message_pb';
-import { shouldBePresent } from '../../../lib/utils/assert/shouldBePresent';
 import { matchDiscriminatedUnion } from '../../../lib/utils/matchDiscriminatedUnion';
-import { Chain } from '../../../model/chain';
 import { BlockchainServiceFactory } from '../../../services/Blockchain/BlockchainServiceFactory';
+import { getKeysignChain } from './getKeysignChain';
 
 type Input = {
   keysignPayload: KeysignPayload;
   walletCore: WalletCore;
 };
 
-export const getPreSignedImageHashes = async ({
+export const getTxInputData = async ({
   keysignPayload,
   walletCore,
-}: Input): Promise<string[]> => {
-  const coin = shouldBePresent(keysignPayload.coin);
-  const chain = coin.chain as Chain;
+}: Input): Promise<Uint8Array[]> => {
+  const chain = getKeysignChain(keysignPayload);
```

Comment:
here isn't you will get two Uint8Array[]?

Reply:
This function takes a `KeysignPayload` as input and returns an array of input data. Typically, it will be an array containing a single `Uint8Array`. However, in the case of a swap with approval, the array will contain two elements. Therefore, the return type is `Uint8Array[]`.